content/appendices.md

```diff
@@ -611,11 +611,20 @@ characters permitted in user ID localparts. There are currently active
 users whose user IDs do not conform to the permitted character set, and
 a number of rooms whose history includes events with a `sender` which
 does not conform. In order to handle these rooms successfully, clients
-and servers MUST accept user IDs with localparts from the expanded
-character set:
+and servers MUST accept user IDs with localparts consisting of any legal
+unicode codepoint except for `:`, including zero characters. Localparts
+MUST be valid UTF-8 sequences.
+
+Servers SHOULD NOT produce user IDs with localparts outside of the following
+character set, and SHOULD NOT forward such user IDs to clients when referenced
+outside the context of an event. For example, device list updates from "invalid"
+user IDs would be dropped by the receiving server.
```

Comment:
This is worse than just removing events by these users, as things will only half work (E2EE will entirely break for them in weird and wonderful ways).

Reply:
If you set your localpart to a smiley poo, E2EE will break for you. That's fine by me. What I care about is you not being able to DoS a room by setting your localpart to a smiley poo, which is why we have to accept events over federation whose senders are invalid; we don't have to do much else to help such users.